ush/python/pygfs/task/marine_analysis.py

```diff
@@ -0,0 +1,490 @@
+#!/usr/bin/env python3
+
+import copy
+import os
+from logging import getLogger
+import pygfs.utils.marine_da_utils as mdau
+import glob
+import re
+import netCDF4
+from multiprocessing import Process
+import subprocess
+import yaml
+from jcb import render
+
+from wxflow import (AttrDict,
+                    FileHandler,
+                    add_to_datetime, to_timedelta, to_YMD,
+                    parse_j2yaml,
+                    logit,
+                    Executable,
+                    Task,
+                    Template, TemplateConstants, YAMLFile)
+
+logger = getLogger(__name__.split('.')[-1])
+
+
+def parse_obs_list_file(obs_list_yaml_path):
+    # Get the list of observation types from the obs_list.yaml
+    obs_types = []
+    with open(obs_list_yaml_path, 'r') as file:
+        for line in file:
+            # Remove leading/trailing whitespace and check if the line is uncommented
+            line = line.strip()
+            if line.startswith('- !INC') and not line.startswith('#'):
+                # Extract the type using regex
+                match = re.search(r'\$\{MARINE_OBS_YAML_DIR\}/(.+)\.yaml', line)
+                if match:
+                    obs_types.append(str(match.group(1)))
+    return obs_types
+
+
+class MarineAnalysis(Task):
+    """
+    Class for global marine analysis tasks
+    """
+    @logit(logger, name="MarineAnalysis")
+    def __init__(self, config):
+        super().__init__(config)
+        _calc_scale_exec = os.path.join(self.task_config.HOMEgfs, 'ush', 'soca', 'calc_scales.py')
+        _window_begin = add_to_datetime(self.task_config.current_cycle, -to_timedelta(f"{self.task_config.assim_freq}H") / 2)
+        _window_end = add_to_datetime(self.task_config.current_cycle, to_timedelta(f"{self.task_config.assim_freq}H") / 2)
+
+        # compute the relative path from self.task_config.DATA to self.task_config.DATAenspert
+        if self.task_config.NMEM_ENS > 0:
+            _enspert_relpath = os.path.relpath(self.task_config.DATAenspert, self.task_config.DATA)
+        else:
+            _enspert_relpath = None
+
+        # Create a local dictionary that is repeatedly used across this class
+        local_dict = AttrDict(
+            {
+                'PARMsoca': os.path.join(self.task_config.PARMgfs, 'gdas', 'soca'),
+                'MARINE_WINDOW_BEGIN': _window_begin,
+                'MARINE_WINDOW_BEGIN_ISO': _window_begin.strftime('%Y-%m-%dT%H:%M:%SZ'),
+                'MARINE_WINDOW_END': _window_end,
+                'MARINE_WINDOW_LENGTH': f"PT{self.task_config['assim_freq']}H",
+                'MARINE_WINDOW_MIDDLE': self.task_config.current_cycle,
+                'MARINE_WINDOW_MIDDLE_ISO': self.task_config.current_cycle.strftime('%Y-%m-%dT%H:%M:%SZ'),
```

Comment:
`wxflow` has a function `to_isotime`
Please use the framework where possible.

Reply:
Will try. I forgot why I did not to this the first time around.

Reply:
That requires too many changes, this will be addressed as part of https://github.com/NOAA-EMC/GDASApp/issues/1296